src/main/java/codeu/model/store/basic/SettingStore.java

```diff
@@ -38,26 +38,35 @@ private SettingStore(PersistentStorageAgent persistentStorageAgent) {
         settings = new ArrayList<>();
     }
     //
-    public void updateColor(Setting setting) {
+    public void updateSetting(Setting setting) {
+        persistentStorageAgent.writeThrough(setting);
+    }
+    /**
+     * Add a new setting to the current set of settings known to the application.
+     */
+    public void addSetting(Setting setting) {
         settings.add(setting);
```

Comment:
I dont think we should have an "addSetting" method. The only place where we should be adding setting is when a user is on app startup. So in PersistentDataStore. The rest of the system should only be updating the setting.

Reply:
Removed.